src/nvim/profile.c

```diff
@@ -944,3 +944,43 @@ void time_msg(const char *mesg, const proftime_T *start)
   g_prev_time = now;
   fprintf(time_fd, ": %s\n", mesg);
 }
+
+/// Initializes the time time_fd stream used to write startup times
+///
+/// @param startup_time_file the startuptime report file path
+/// @param process_name the name of the current process to write in the report.
+void time_init(const char *startup_time_file, const char *process_name)
+{
+  time_fd = fopen(startup_time_file, "a");
+  if (time_fd == NULL) {
+    semsg(_(e_notopen), startup_time_file);
+    return;
+  }
+  startup_time_buf = xmalloc(sizeof(char) * (STARTUP_TIME_BUF_SIZE + 1));
+  // set a large buffer so that flushing only happens after initialization
+  if (setvbuf(time_fd, startup_time_buf, _IOFBF, STARTUP_TIME_BUF_SIZE + 1) != 0) {
```

Comment:
Is there any more insight that can be given in the comment here about how `setvbuf` is being leveraged? It's somewhat subtle (and not common in the codebase).

Reply:
tried to give more information in the comment